File: Collections/Models/VaultKeep.cs
namespace Collections.Models
{
  public class VaultKeep
  {
    public int Id { get; set; }
    public string CreatorId { get; set; }
    public int VaultId { get; set; }
    public int KeepId { get; set; }
    public Profile Creator {get; set;}
    public Keep Keep { get; set; }
    public Vault Vault { get; set; }
  }
}

File: Collections.client/src/models/VaultKeep.js
export class VaultKeep {
  constructor(vaultKeepData = {}) {
    this.id = vaultKeepData.id
    this.creatorId = vaultKeepData.creatorId
    this.vaultId = vaultKeepData.vaultId
    this.keepId = vaultKeepData.keepId
    this.creator = vaultKeepData.creator
    this.keep = vaultKeepData.keep
  }
}